readthedocs/projects/tasks.py

```diff
@@ -564,24 +564,33 @@ def run_build(self, docker, record):
                 # TODO the build object should have an idea of these states,
                 # extend the model to include an idea of these outcomes
                 outcomes = self.build_docs()
-                build_id = self.build.get('id')
             except vcs_support_utils.LockTimeout as e:
                 self.task.retry(exc=e, throw=False)
                 raise VersionLockedError
             except SoftTimeLimitExceeded:
                 raise BuildTimeoutError
-
-            # Finalize build and update web servers
-            if build_id:
-                self.update_app_instances(
-                    html=bool(outcomes['html']),
-                    search=bool(outcomes['search']),
-                    localmedia=bool(outcomes['localmedia']),
-                    pdf=bool(outcomes['pdf']),
-                    epub=bool(outcomes['epub']),
-                )
             else:
```

Comment:
Is this `else` on the `try` instead of the `if` now? I can never remember what `try/else` does.

Reply:
That's correct. An `else` on a `try` is taken if there is no exception. I refactored that slightly because we defined `build_id` in a `try` block and then use it later. The only reason it wasn't ever a `NameError` is that we reraise the error in both `except` blocks.